opsyclient/shell.py

```diff
@@ -46,15 +63,301 @@ def zones_cli():
+    pprint(client.zones.list_zones(**set_args).response().result)
+
+
+@zones_cli.command('add')
+@click.argument('zone_name')
+@click.pass_obj
+def zones_add(client, zone_name):
+    """Create a new zone."""
+    try:
+        pprint(client.zones.create_zone(zone_name).reponse().result)
+    except HTTPError as err:
+        error_handler('add zone', zone_name, err.code, err.reason)
+
+
+@zones_cli.command('delete')
+@click.argument('zone_name_or_id')
+@click.pass_obj
+def zones_delete(client, zone_name_or_id):
+    """Delete a zone."""
+    try:
+        pprint(client.zones.delete_zone(zone_name_or_id).response().result)
+    except HTTPError as err:
+        error_handler('delete zone', zone_name_or_id, err.code, err.reason)
+
+
+@zones_cli.command('update')
+@click.argument('zone_name_or_id')
+@click.option('--name', '-n', help="Zone name.")
+@click.option('--description', '-d', help="Zone description.")
+@click.option('--vars', '-v', help="A dictionary of zone vars.",
+              callback=validate_json)
+@click.pass_obj
+def zones_update(client, zone_name_or_id, **kwargs):
+    """Update a zone."""
+    set_args = {k: v for k, v in kwargs.items() if v is not None}
+    try:
+        pprint(client.zones.update_zone(zone_name_or_id, **set_args).response().result)
+    except HTTPError as err:
+        error_handler('update zone', zone_name_or_id, err.code, err.reason)
+
+
+################################################################################
+## Host Group ##
+################################################################################
+@cli.group('host')
+def host_cli():
+    """Commands related to hosts."""
+
+
+@host_cli.command('show')
+@click.argument('host_name_or_id')
+@click.pass_obj
+def host_show(client, host_name_or_id):
+    """Show a host."""
+    try:
+        pprint(client.hosts.show_host(host_name_or_id).response().result)
+    except HTTPError as err:
+        error_handler('show host', host_name_or_id, err.code, err.reason)
+
+
+@host_cli.command('list')
+@click.option('--id', help="Host ID to filter on.")
+@click.option('--zone-id', help="Zone ID to filter on.")
+@click.option('--zone-name', help="Zone name to filter on.")
+@click.option('--group-id', help="Group ID to filter on.")
+@click.option('--group-name', help="Group name to filter on.")
+@click.option('--name', help="Host name to filter on.")
+@click.option('--device-id', help="Device ID to filter on.")
+@click.option('--bmc-ip', help="BMC IP address to filter on.")
+@click.pass_obj
+def host_list(client, **kwargs):
+    """List all hosts, filterable."""
+    set_args = {k: v for k, v in kwargs.items() if v is not None}
+    pprint(client.hosts.list_hosts(**set_args).response().result)
+
+
+@host_cli.command('add')
+@click.argument('host_name')
+@click.argument('zone_id')
+@click.option('--device-id', '-d', help="External device identifier.")
+@click.option('--bmc-ip', '-b', help="IP of the BMC (aka DRAC/iLo/etc).")
```

Comment:
We should probably have an IP validator for this. We'd only want the validator on the add and update, not on list since that can accept filtering special characters (`*` and `!` for example).